src/arduino/ports/samd/tusb_config_samd.h

```diff
@@ -38,7 +38,7 @@ extern "C" {
 #define CFG_TUSB_MCU OPT_MCU_SAMD21
 #endif
 
-#ifdef USE_TINYUSB
+#if defined(USE_TINYUSB) || !defined(USBCON)

```

Comment:
can you explain again, why putting `|| !defined(USBCON)` would help. I can imagine you drop the USBCON in your board.txt, why not also add USE_TINYUSB as well. I would prefer USE_TINYUSB is explicitly specify when using with the stack.

Reply:
In my proposed changes to the Arduino SAMD core USBCON would not longer be defined as you have said. I don't add USE_TINYUSB because it would be a library specific change to the Arduino core and I think the team would be less likely to accept it. 

Also, if USE_TINYUSB was added, [the logic for defining Serial](https://github.com/adafruit/Adafruit_TinyUSB_Arduino/blob/9527382b498cd1264d9132f9e7855434c869f557/src/arduino/Adafruit_USBD_CDC.h#L91) would need updated as it would conflict with the Arduino SAMD core's definition. This is another reason why I chose this route.

I agree that not explicitly specifying USE_TINYUSB is not ideal but you've said yourself that Arduino team probably would not want to do it.

Reply:
from Arduino core pov, `USBCON` is defined when using CDC as Serial, and there is other board which use FTDI/CP210x as Serial, which lack of `USBCON` actually means. It may not be popular with samd core, but it is with avr one. And `USBCON` is consistently used as that one. 

Therefore I don't think you could persuade Arduino team to use lack of it as mean for TinyUSB, especially Arduino goal is having consistent API across other cores and libraries. Asking them to add menu for other usb stack has a better standing ground IMHO.  But let wait to see if I am wrong.

Reply:
That makes sense. I haven't looked much at the AVR core so I didn't have that experience for reference. I figured USBCON had a more important use but like you said it wasn't obvious within the SAMD core.

If they aren't comfortable with it, a different defiine can be created to disable the built-in USB core. 

Ideally a menu option could be added for TinyUSB but we'll see.